components/SearchResult.tsx

```diff
@@ -1,32 +1,43 @@
+"use client";
 import React from "react";
 import { useQuery } from "@tanstack/react-query";
 import { SearchResponseSchema } from "@/app/utils/discogs.schemas";
+import SearchBar from "./SearchBar";
 
-export default function SearchResult({ search }: { search: string }) {
-  const { data, isError, isLoading } = useQuery({
-    queryKey: ["search"],
-    queryFn: async () => {
-      const params = new URLSearchParams({ search: "the strokes" });
-      const result = await fetch("/api/search?" + params).then((res) =>
-        res.json()
-      );
-      const parsedResult = SearchResponseSchema.parse(result);
-      return parsedResult;
-    },
-  });
+export default function Search() {
+  const [searchValue, setSearchValue] = React.useState<string>("");
 
-  console.log("data: ", data);
+  function SearchResult() {
```

Comment:
you never want to define components within components like this.

you have the right idea though, just put this function/component in a different file

Reply:
you'll have to pass `searchValue` as a prop to it though